google/cloud/aiplatform/featurestore/entity_type.py

```diff
@@ -1146,6 +1149,148 @@ def ingest_from_gcs(
+                     feature_source_fields = None or {}
+
+                     In case all features' source field and ID do not match:
+                     feature_source_fields = {
+                        'my_feature_id_1': 'my_feature_id_1_source_field',
+                        'my_feature_id_2': 'my_feature_id_2_source_field',
+                        'my_feature_id_3': 'my_feature_id_3_source_field',
+                     }
+
+                     In case some features' source field and ID do not match:
+                     feature_source_fields = {
+                        'my_feature_id_1': 'my_feature_id_1_source_field',
+                     }
+            entity_id_field (str):
+                Optional. Source column that holds entity IDs. If not provided, entity
+                IDs are extracted from the column named ``entity_id``.
+            disable_online_serving (bool):
+                Optional. If set, data will not be imported for online
+                serving. This is typically used for backfilling,
+                where Feature generation timestamps are not in
+                the timestamp range needed for online serving.
+            worker_count (int):
+                Optional. Specifies the number of workers that are used
+                to write data to the Featurestore. Consider the
+                online serving capacity that you require to
+                achieve the desired import throughput without
+                interfering with online serving. The value must
+                be positive, and less than or equal to 100. If
+                not set, defaults to using 1 worker. The low
+                count ensures minimal impact on online serving
+                performance.
+            request_metadata (Sequence[Tuple[str, str]]):
+                Optional. Strings which should be sent along with the request as metadata.
+
+        Returns:
+            EntityType - The entityType resource object with feature values imported.
+
+        """
+        try:
+            import pyarrow  # noqa: F401 - skip check for 'pyarrow' which is required when using 'google.cloud.bigquery'
+        except ImportError:
+            raise ImportError(
+                f"Pyarrow is not installed. Please install pyarrow to use "
+                f"{self.ingest_from_df.__name__}"
+            )
+
+        bigquery_client = bigquery.Client(
+            project=self.project, credentials=self.credentials
+        )
+
+        entity_type_name_components = self._parse_resource_name(self.resource_name)
+        featurestore_id, entity_type_id = (
+            entity_type_name_components["featurestore"],
+            entity_type_name_components["entity_type"],
+        )
+
+        temp_bq_dataset_name = f"temp_{featurestore_id}_{uuid.uuid4()}".replace(
+            "-", "_"
+        )
+        temp_bq_dataset_id = f"{initializer.global_config.project}.{temp_bq_dataset_name}"[
```

Comment:
`self.project` returns the project number, but bq dataset need a project ID

Reply:
Got it. This is an issue moving forward because the project used to construct `EntityType` might not match the project ID set in `init` or, if project is not set using `init`, the project ID returned from `google.auth.default`.

Additionally, we don't stop the user from using a project number when setting `init`.

We shouldn't block this PR as there is a workaround but we should capture this issue in a ticket.

Reply:
sounds good, will address this in a separate ticket/PR